components/tabs/src/storage.rs

```diff
@@ -131,8 +155,11 @@ impl TabsStorage {
             flags,
             &crate::schema::TabsMigrationLogic,
         )?;
-        self.db_connection = Some(conn);
-        Ok(self.db_connection.as_ref().unwrap())
+        self.db_connection = DbConnection::Open(conn);
+        match self.db_connection {
+            DbConnection::Open(ref conn) => Ok(conn),
+            _ => Err(Error::UnexpectedConnectionState),
```

Comment:
This might be a good place for `unreachable!()`. Since we just moved the connection into `self.db_connection` above, and we know we have exclusive access to the `&mut self`, any other variant here is impossible.